app/helpers/tabs_helper.rb

```diff
@@ -22,7 +22,12 @@ def tabs
     Edition::Tab.all
   end
 
-  def tabs_for(user)
-    tabs.reject { |tab| tab.name == "unpublish" unless user.govuk_editor? }
+  def tabs_for(user, resource)
+    tabs_to_remove = [
+      user.has_editor_permissions?(resource) ? "" : "admin",
+      user.govuk_editor? ? "" : "unpublish",
+    ]
```

Comment:
This is a little bit of a smell to me, as we could end up with `tabs_to_remove == ["", ""]`, which will work but is a bit surprising. I think I'd opt for a different approach, e.g.

```ruby
tabs_to_remove = []
tabs_to_remove << "admin" unless user.has_editor_permissions?(resource)
tabs_to_remove << "unpublish" unless user.govuk_editor?
```

Reply:
I did this, and then changed my mind for some reason. thanks!